ocs_ci/ocs/version.py

```diff
@@ -0,0 +1,179 @@
+
+    logger.info("found storage namespaces %s", storage_namespaces)
+
+    # Now get the OCS version by asking for version of all container images of
+    # all pods in openshift-storage namespace.
+    image_dict = {}
+    for ns in storage_namespaces:
+        ocs = OCP(kind="pod", namespace=ns)
+        ocs_pods = ocs.get()
+        ns_dict = {}
+        for pod in ocs_pods['items']:
+            for container in pod['spec']['containers']:
+                logger.debug(
+                    'container %s (in pod %s) uses image %s',
+                    container['name'],
+                    pod['metadata']['name'],
+                    container['image'])
+            cs_items = pod['status'].get('containerStatuses')
+            if cs_items is None:
+                pod_name = pod['metadata']['name']
+                logger.warning(f'pod {pod_name} has no containerStatuses')
+                continue
+            for cs in cs_items:
+                ns_dict.setdefault(cs['image'], set()).add(cs['imageID'])
+        image_dict.setdefault(ns, ns_dict)
+
+    logger.debug("ocs version collected")
+
+    return cluster_version, image_dict
+
+
+def report_ocs_version(cluster_version, image_dict, file_obj):
+    """
+    Report OCS version via:
+
+     * python logging
+     * printing human readable version into file_obj (stdout by default)
+    """
+    # log the version
+    logger.info("ClusterVersion .spec.channel: %s", cluster_version["spec"]["channel"])
+    logger.info("ClusterVersion .status.desired.version: %s", cluster_version["status"]["desired"]["version"])
+    logger.info("ClusterVersion .status.desired.image: %s", cluster_version["status"]["desired"]["image"])
+
+    # write human readable version of the above
+    print(f'cluster channel: {cluster_version["spec"]["channel"]}', file=file_obj)
+    print(f'cluster version: {cluster_version["status"]["desired"]["version"]}', file=file_obj)
+    print(f'cluster image: {cluster_version["status"]["desired"]["image"]}', file=file_obj)
+
+    for ns, ns_dict in image_dict.items():
+        logger.info("storage namespace %s", ns)
+        print(f"\nstorage namespace {ns}", file=file_obj)
+        for image, image_ids in ns_dict.items():
+            logger.info("image %s %s", image, image_ids)
+            print(f"image {image}", file=file_obj)
+            # TODO: should len(image_ids) == 1?
+            if len(image_ids) > 1:
+                logging.warning("there are at least 2 different imageIDs for image %s", image)
+            for image_id in image_ids:
+                print(f" * {image_id}", file=file_obj)
+
+
+def main():
+    """
+    Main fuction of version reporting command line tool.
+
+    Based on suggestion from
+    https://github.com/red-hat-storage/ocs-ci/issues/356
+    """
+    ap = argparse.ArgumentParser(
+        description="report OCS version for QE purposes")
+    ap.add_argument(
+        "--cluster-path",
```

Comment:
I am bit confused on --cluster-path to report OCS version, do you need this option?

Reply:
Yes. This way you specify which cluster do you like to get a version report of, also it's used to get credentials for the script to be able to access the cluster.

Reply:
I see the use case now. thanks